File: examples/test_env.py
#!/usr/bin/env python3
"""
Script de test pour vérifier l'injection des secrets via le SDK lazy-locker.

Usage:
  1. Lancez lazy-locker et entrez votre passphrase (l'agent démarre automatiquement)
  2. python test_env.py (ou uv run test_env.py, bun run test_env.py, etc.)
"""

import os
import sys

# Add SDK to path for local testing
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', 'sdk', 'python'))

from lazy_locker import inject_secrets, is_agent_running, status

print("=" * 50)
print("Test du SDK lazy-locker pour Python")
print("=" * 50)

# Vérifier si l'agent est actif
if not is_agent_running():
    print("\n❌ Agent non démarré!")
    print("   Lancez 'lazy-locker' et entrez votre passphrase.")
    sys.exit(1)

# Afficher le statut
try:
    info = status()
    remaining_hours = info.get('ttl_remaining_secs', 0) // 3600
    remaining_mins = (info.get('ttl_remaining_secs', 0) % 3600) // 60
    print(f"\n✅ Agent actif (TTL: {remaining_hours}h {remaining_mins}m)")
except Exception as e:
    print(f"\n⚠️  Erreur statut: {e}")

# Injecter les secrets
try:
    count = inject_secrets()
    print(f"✅ {count} secrets injectés dans os.environ")
except Exception as e:
    print(f"\n❌ Erreur injection: {e}")
    sys.exit(1)

# Afficher les secrets (masqués)
print("\nSecrets disponibles:")
for key in ['test', 'test2', 'MY_API_KEY', 'DB_PASSWORD']:
    value = os.getenv(key)
    if value:
        masked = value[:3] + '*' * (len(value) - 3) if len(value) > 3 else '***'
        print(f"  {key} = {masked}")
    else:
        print(f"  {key} = (non défini)")

print("\n" + "=" * 50)
print("✅ Test réussi! Les secrets sont injectés.")
print("=" * 50)

File: examples/test_env.ts
/**
 * Script de test pour vérifier l'injection des secrets via le SDK lazy-locker.
 *
 * Usage:
 *   1. Lancez lazy-locker et entrez votre passphrase (l'agent démarre automatiquement)
 *   2. bun run test_env.ts (ou npx tsx test_env.ts)
 */

import { injectSecrets, isAgentRunning, status, getSecrets } from '../sdk/javascript/src/index';

console.log('='.repeat(50));
console.log('Test du SDK lazy-locker pour JavaScript/TypeScript');
console.log('='.repeat(50));

async function main() {
  // Vérifier si l'agent est actif
  if (!(await isAgentRunning())) {
    console.log('\n❌ Agent non démarré!');
    console.log("   Lancez 'lazy-locker' et entrez votre passphrase.");
    process.exit(1);
  }

  // Afficher le statut
  try {
    const info = await status();
    const remainingHours = Math.floor(info.ttl_remaining_secs / 3600);
    const remainingMins = Math.floor((info.ttl_remaining_secs % 3600) / 60);
    console.log(`\n✅ Agent actif (TTL: ${remainingHours}h ${remainingMins}m)`);
  } catch (e) {
    console.log(`\n⚠️  Erreur statut: ${e}`);
  }

  // Injecter les secrets
  try {
    const count = await injectSecrets();
    console.log(`✅ ${count} secrets injectés dans process.env`);
  } catch (e) {
    console.log(`\n❌ Erreur injection: ${e}`);
    process.exit(1);
  }

  // Afficher les secrets (masqués)
  console.log('\nSecrets disponibles:');
  const secrets = await getSecrets();
  for (const [key, value] of Object.entries(secrets)) {
    const masked = value.length > 3 ? value.slice(0, 3) + '*'.repeat(value.length - 3) : '***';
    console.log(`  ${key} = ${masked}`);
  }

  console.log('\n' + '='.repeat(50));
  console.log('✅ Test réussi! Les secrets sont injectés.');
  console.log('='.repeat(50));
}

main().catch(console.error);
